src/Calculator/Views/MainPage.xaml.cs

```diff
@@ -61,24 +62,38 @@ public MainPage()
 
             UserActivityRequestManager.GetForCurrentView().UserActivityRequested += async (_, args) =>
             {
-                var deferral = args.GetDeferral();
-                if (deferral == null)
+                using (var deferral = args.GetDeferral())
                 {
-                    // Windows Bug in ni_moment won't return the deferral propoerly, see https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/47775705/
-                    return;
+                    if (deferral == null)
+                    {
+                        // FIXME: https://microsoft.visualstudio.com/DefaultCollection/OS/_workitems/edit/47775705/
+                        TraceLogger.GetInstance().LogRecallError("55e29ba5-6097-40ec-8960-458750be3039");
+                        return;
+                    }
+                    var channel = UserActivityChannel.GetDefault();
+                    var activity = await channel.GetOrCreateUserActivityAsync($"{Guid.NewGuid()}");
+                    string embeddedData;
+                    try
+                    {
+                        var json = JsonSerializer.Serialize(new ApplicationSnapshotAlias(Model.Snapshot));
```

Comment:
Perhaps we can use `JsonSerializer.SerializeToUtf8Bytes` to save the transforming work?

Reply:
probably next time when we need it. because 1) it's not a bottle neck of performance here. 2)`JsonSerializer` doesn't have a good deserializer accepting utf-8 bytes. 3) it's easy to debug with string rather than an array of bytes.